route/build/src/historyFile.f90

```diff
@@ -34,7 +34,7 @@ MODULE historyFile
     integer(i4b)          :: iTime=0             ! time step in output netCDF
     logical(lgt)          :: fileStatus=.false.  ! flag to indicate history output netcdf is open
     logical(lgt)          :: gageOutput=.false.  ! flag to indicate this is at-gage-only output (== output subset of reaches)
-    type(iosystem_desc_t) :: pioSystem           ! PIO system (this does not have to be initialize for each history file?)
```

Comment:
reason why I changed `pioSystem` to `pioSys` is `pioSystem` is global variable

Reply:
So it's being passed down by argument rather than accessing a global variable from a module? That makes sense.

I wonder about this sort of thing as well. Even though it's passed down it actually IS the global variable. So you could argue you shouldn't change the name.

Another option would be to do something like PioSystemLocal -- to note that it's a local variable. I agree I'm not sure the best way to handle this either...

Reply:
For data within derived data type, variable name can be the same as global variable.  `histFile%pioSystem `and `pioSystem` are different. but just my preference (not use the same name). For argument variable name for procudre, they become confusing (and not sure how the compiler handle (but this is not the case here).